### PR TITLE
php73: fix builds on darwin architecture

### PR DESCRIPTION
**Pull request by drupol**
This PR:

- [x] Is related to #200 
- [x] Add a new patch for PHP 7.3 on Darwin architecture

### CODE REVIEW

**Comment by HelloWorld017** [comment]
I'll check whether this builds well on my machine

**Comment by HelloWorld017** [comment]
This derivation also keeps failing with following issue:
```
PEAR package PHP_Archive not installed: generated phar will require PHP's phar extension be enabled.
```

The flag `--with-pcre-regex=${prev.pcre2.dev}` should be added in order to the php73 built.
Here's the commit:
```
$ git am - <<'EOF'
From ac13f9d341dc63bf366699cfc2911601b73a849b Mon Sep 17 00:00:00 2001
From: nenw <khi@nenw.dev>
Date: Fri, 20 Jan 2023 20:00:17 +0900
Subject: [PATCH] php73: fix pcre config flags

---
 pkgs/phps.nix | 1 +
 1 file changed, 1 insertion(+)

diff --git a/pkgs/phps.nix b/pkgs/phps.nix
index 71a884d..760397e 100644
--- a/pkgs/phps.nix
+++ b/pkgs/phps.nix
@@ -43,6 +43,7 @@ let
 
               "--enable-libxml"
               "--with-libxml-dir=${prev.libxml2.dev}"
+              "--with-pcre-regex=${prev.pcre2.dev}"
             ];
 
           buildInputs =
-- 
2.38.1

EOF
```

**Comment by drupol** [comment]
Can you try again now?

**Comment by drupol** [comment]
Can you try again now?

File: pkgs/php/7.3.nix [hunk 1]
```diff
@@ -13,6 +13,9 @@ let
           url = "https://github.com/NixOS/nixpkgs/raw/42e9a2ccfab2a96d28c3c164a6cf41fb6f769de5/pkgs/development/interpreters/php/php73-darwin-isfinite.patch";
           sha256 = "V0mLLmXa2qJyxIVW/7nEml6cXZTBbr42kkJiij9KPyk=";
         })
+        # Fix builds
+        # See https://github.com/fossar/nix-phps/issues/200
```
Comment (jtojnar) [inline pkgs/php/7.3.nix:17]: Maybe use

```nix
# Handle macos versions that don't start with 10.* in libtool.
# https://github.com/php/php-src/commit/d016434ad33284dfaceb8d233351d34356566d7d
(fetchpatch2 {
  url = "https://github.com/php/php-src/commit/d016434ad33284dfaceb8d233351d34356566d7d.patch"
  sha256 = "";
  includes = [
    "build/libtool.m4"
  ];
})
```

Comment (drupol) [reply pkgs/php/7.3.nix]: Done.

**Comment by drupol** [comment]
I pushed a temporary commit (part of #85) to check if it is correctly building on MacOS.

File: pkgs/patches/php73-libtool.patch [hunk 2]
```diff
@@ -0,0 +1,20 @@
+diff -ru a/build/libtool.m4 b/build/libtool.m4
```
Comment (jtojnar) [inline pkgs/patches/php73-libtool.patch:1]: This file can be dropped.

Comment (drupol) [reply pkgs/patches/php73-libtool.patch]: Done.

**Comment by drupol** [comment]
@HelloWorld017 @shyim Any update here?

**Comment by shyim** [comment]
```
shyim in nix-phps on  php73/fix-builds [?] took 5m47s
❯ ./result/bin/php -v
PHP 7.3.28 (cli) (built: Jan 25 2023 08:29:17) ( NTS )
Copyright (c) 1997-2018 The PHP Group
Zend Engine v3.3.28, Copyright (c) 1998-2018 Zend Technologies
    with Zend OPcache v7.3.28, Copyright (c) 1999-2018, by Zend Technologies
```

whooo works

**Comment by HelloWorld017** [comment]
Yes, it works on my laptop too. :+1:

**Comment by drupol** [comment]
@jtojnar Should I remove the commit 2ca5c8717b3b84dd8282877008dbc33801e419a1 before merging? WDYT ?

**Comment by jtojnar** [comment]
Might be faster to merge it that way, since we would have to review that patch otherwise.

**Comment by drupol** [comment]
I just removed the commit.

**Review by jtojnar — APPROVED** [review]

File: pkgs/phps.nix [hunk 3]
```diff
@@ -43,6 +43,9 @@ let
 
               "--enable-libxml"
               "--with-libxml-dir=${prev.libxml2.dev}"
+            ]
+            ++ prev.lib.optionals (prev.lib.versionAtLeast args.version "7.3") [
+              "--with-pcre-regex=${prev.pcre2.dev}"
```
Comment (jtojnar) [inline pkgs/phps.nix:48]: Actually, not sure about this. Should not it be `versionOlder args.version "7.4"`? I would expect newer PHP versions to pick it up using pkg-config.

Comment (drupol) [reply pkgs/phps.nix:48]: I'm not sure either, maybe @shyim or @HelloWorld017 knows?

Comment (drupol) [reply pkgs/phps.nix:48]: Dammit...

Comment (drupol) [reply pkgs/phps.nix:48]: I'll open a new PR right now. Sorry about that.

**Pull request merged**